app.js

```diff
@@ -54,221 +54,81 @@ app.use(async (ctx, next) => {
-    const accountId = ctx.params.accountId;
-    const phoneNumber = ctx.params.phoneNumber;
-
-    const securityCode = password.randomPassword({ length: SECURITY_CODE_DIGITS, characters: password.digits });
-    const [account] = await models.Account.findOrCreate({ where: { accountId, phoneNumber } });
-    await account.update({ securityCode });
-    // TODO: Add code expiration for improved security
-    await sendSecurityCode(account);
-
-    ctx.body = {};
-});
-
-const nacl = require('tweetnacl');
-const crypto = require('crypto');
-const bs58 = require('bs58');
-const verifySignature = async (nearAccount, data, signature) => {
-    try {
-        const hash = crypto.createHash('sha256').update(data).digest();
-        const accessKeys = (await nearAccount.getAccessKeys())
-            .filter(k => k.access_key.permission === 'FullAccess');
-        return accessKeys.some(it => {
-            const publicKey = it.public_key.replace('ed25519:', '');
-            return nacl.sign.detached.verify(hash, Buffer.from(signature, 'base64'), bs58.decode(publicKey));
-        });
-    } catch (e) {
-        console.error(e);
-        return false;
-    }
-};
-
-async function recoveryMethodsFor(account) {
-    if (!account) return [];
-
-    return await account.getRecoveryMethods({
-        attributes: ['createdAt', 'detail', 'kind', 'publicKey']
-    }).map(m => m.toJSON());
-}
-
 router.post('/account/recoveryMethods', checkAccountOwnership, async ctx => {
     const { accountId } = ctx.request.body;
     const account = await models.Account.findOne({ where: { accountId } });
     ctx.body = await recoveryMethodsFor(account);
 });
 
-async function withAccount(ctx, next) {
-    const { accountId } = ctx.request.body;
-    ctx.account = await models.Account.findOne({ where: { accountId } });
-    if (ctx.account) {
-        await next();
-        return;
-    }
-    ctx.throw(404, `Could not find account with accountId: '${accountId}'`);
-}
-
-const recoveryMethods = ['email', 'phone', 'phrase'];
-
-async function checkRecoveryMethod(ctx, next) {
-    const { recoveryMethod } = ctx.request.body;
-    if (recoveryMethods.includes(recoveryMethod)) {
-        await next();
-        return;
-    }
-    ctx.throw(400, `Given recoveryMethod '${recoveryMethod}' invalid; must be one of: ${recoveryMethods.join(', ')}`);
-}
-
 router.post(
     '/account/deleteRecoveryMethod',
     withAccount,
     checkRecoveryMethod,
     checkAccountOwnership,
     withPublicKey,
     async ctx => {
-        const [recoveryMethod] = await ctx.account.getRecoveryMethods({
-            where: {
-                kind: ctx.request.body.recoveryMethod,
-                publicKey: ctx.publicKey,
-            }
-        });
+        const { kind, publicKey } = ctx.request.body;
+        const [recoveryMethod] = await ctx.account.getRecoveryMethods({ where: {
+            kind: kind,
+            publicKey: publicKey,
+        }});
```

Comment:
I would prefer that style changes like this come from a linter and get added all at once; this PR is already going to be pretty big without a bunch of style-change noise